src/github.com/getlantern/flashlight/pac.go

```diff
@@ -41,16 +44,33 @@ func setUpPacTool() error {
 	return nil
 }
 
+func setProxyAddr(addr string) {
```

Comment:
<10ms in average. See simple benchmark [here]https://github.com/getlantern/lantern/issues/2099#issuecomment-78971159

If we choose auto-detection, we have to pass all traffic through Lantern.